### ISSUE TITLE
Kea does not send hostnames in DHCP replies

### ISSUE THREAD

**Issue by creideiki** [saab-simc-admin/palletjack#54]
When configured using `palletjack2kea`, Kea's DHCP replies do not include the client hostname, even though it's included in `kea.conf`.

Tested on both kea-1.0.0-9.el7.x86_64 from EPEL and a hand-built Kea 1.1.0-beta.

**Comment by creideiki** [comment]
Things which do not work:
- Including disabled dynamic DNS update configuration with a domain name: no difference

``` json
    "dhcp-ddns": {
      "enable-updates": false,
      "qualifying-suffix": "example.com"
    }
```
- Including enabled dynamic DNS update configuration with a domain name: no difference

``` json
    "dhcp-ddns": {
      "enable-updates": true,
      "qualifying-suffix": "example.com"
    }
```
- Not using the `hostname` configuration key, but instead explicitly constructing an option 12 record: no such option defined.
- Defining a custom option 12 from [the spec](https://tools.ietf.org/html/rfc1533#section-3.14): cannot redefine standard option.
- Removing lease database `/var/lib/kea/kea-leases4.csv`, just in case it's cached the absence of a hostname: no difference

**Comment by creideiki** [comment]
... and then I left it running for a couple of hours, and when I got back it seems to have started sending out hostnames anyway. Huh?

This is a hand-built Kea 1.1.0-beta, with configuration:

``` yaml
"dhcp-ddns": {
  "enable-updates": true,
  "qualifying-suffix": "example.com"
}
```

**Comment by creideiki** [comment]
It seems the replies only contain hostnames when the client includes them in its DHCPREQUEST messages.

**Comment by creideiki** [comment]
The [host reservation design document](https://kea.isc.org/wiki/HostReservationDesign) says:

> If the client is requesting an update to DDNS, the server will send back the name "super-host.isc.org" in the Client FQDN option.

I wonder how it does that.

**Comment by creideiki** [comment]
[This bug](http://kea.isc.org/ticket/5005) was closed in Kea 1.1.0 final. Built it from source, and it works.

**Comment by creideiki** [comment]
Kea 1.1.0 is in EPEL testing.
